src/color.rs

```diff
@@ -420,6 +420,153 @@ impl Color {
+    /// use turtle::Color;
+    /// 
+    /// let c1 = Color::rgba(18., 55., 125., 1.0);
+    /// // This will panic
+    /// let mixed = c1.mix(&Color::rgba(125., 33., 200., 0.8), 101);
+    /// ```
+    pub fn mix(&self, with: &Color, weight: u32) -> Self {
+        if weight > 100 {
+            assert!(weight <= 100, "{} is not a valid value for weight. Must be within 0-100 inclusive", weight);
+        }
+        // This algorighm cribbed from Sass (http://sass-lang.com/documentation/Sass/Script/Functions.html#mix-instance_method)
+        // with some modifications to use floating point facilities provided by Rust
+        let p = weight as f64 / 100.0;
+        let w = p.mul_add(2., -1.);
+        let a = self.alpha - with.alpha;
+
+        let w1 = if w * a == -1. {
+            (w + 1.) / 2.
+        } else {
+            ((w + a) / a.mul_add(w, 1.) + 1.) / 2.
+        };
+
+        let w2 = 1. - w1;
+
+        let r_mod = self.red.mul_add(w1, with.red.mul_add(w2, 0.)).round();
+        let g_mod = self.green.mul_add(w1, with.green.mul_add(w2, 0.)).round();
+        let b_mod = self.blue.mul_add(w1, with.blue.mul_add(w2, 0.)).round();
+        let a_mod = self.alpha * p + with.alpha * (1. - p);
+        Color::rgba(r_mod, g_mod, b_mod, a_mod)
+    }
+
+    /// Retrieve the hue for this `Color`. The returned value is in degrees 
+    /// between 0° and 360° that represents its position on the color wheel. 
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// 
+    /// let c: Color = "blue".into();
+    /// assert_eq!(201.0, c.hue());
+    /// ```
+    pub fn hue(&self) -> f64 {
+        self.to_hsl().0
+    }
+
+    /// Retrieve the saturation value for this `Color`. The returned value is 
+    /// between 0.0 and 1.0 (inclusive) that indicates the saturation percentage.
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// 
+    /// let c: Color = "blue".into();
+    /// assert_eq!(1.0, c.saturation());
+    /// ```
+    pub fn saturation(&self) -> f64 {
+        self.to_hsl().1
+    }
+
+    /// Retrieve the lightness value for this `Color`. The returned value is between
+    /// 0.0 and 1.0 (inclusive) that indicates the lightness percentage.
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// 
+    /// let c: Color = "blue".into();
+    /// assert_eq!(0.39215686274509803, c.lightness());
+    /// ```
+    pub fn lightness(&self) -> f64 {
+        self.to_hsl().2
+    }
+
+    /// Create a new `Color` by shifting the hue of this `Color` by the specified amount.
+    /// The hue is specified in degrees, ranging from -360 to 360 degrees.
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// 
+    /// // Positive values
+    /// let c: Color = "orange".into();
+    /// assert_eq!(c.with_hue(70.), Color::rgb(130.0, 245.0, 48.0));
+    /// 
+    /// // Negative values
+    /// assert_eq!(c.with_hue(-70.), Color::rgb(245.0, 48.0, 196.0));
+    /// ```
+    pub fn with_hue(&self, hue: f64) -> Self {
```

Comment:
Please take `self` by value (use `self` instead of `&self`) in this method and all of the methods that you have added in this PR.

Reply:
Should we check that the value of `hue` is within its acceptable range?

Reply:
This was a toss-up for range checking. Since we can mod the values to normalize, allowing multiple rotations around the color wheel won't really hurt. And most implementations I found (such as `sass`) do not exhibit any issues with values outside the range of -360 and +360. That being said, there is some argument to be made for the extra clarity in restricting the range. Personally, I feel that not restricting the value keeps this implementation more in line with other existing implementations, allowing an easy transfer of existing knowledge or porting of code. Whether those cases are actually going to happen however, I can't really say.

Reply:
>Please take self by value (use self instead of &self) in this method and all of the methods that you have added in this PR.

I assume that we should still retain the reference receiver for `hue`, `saturation`, and `lightness` methods as they are simply getters and consuming the underlying `Color` would probably not be the intended behavior in this case.

Reply:
If you look at other Copy types (e.g. [f64](https://doc.rust-lang.org/std/primitive.f64.html)), all methods (even simple getters) take self by value. I think we should do the same here.

Reply:
I am okay with not validating hue since it doesn't matter but I wonder if there would ever be a case where a value like 456° or something would be confusing because it falls outside the range and yet still works. Maybe we should validate and panic just because that would create the least amount of surprise. Another option is to just not have a limited value for that at all. If there's no reason that the value should be within a certain range, we shouldn't describe it as such. If so, we could say something like "the value typically falls between 0° and 360°. Values outside of that range will be coerced into it using the modulus operator."

Reply:
It seems the cleanest option is to set an established a range of -360 to 360 and validate it in the code with an descriptive panic value. I'll add that and create a couple tests.